runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectRunnerTest.java

```diff
@@ -87,15 +93,59 @@
  */
 @RunWith(JUnit4.class)
 public class DirectRunnerTest implements Serializable {
+  private static final Map<String, Context> TEAR_DOWN_CALLED = new HashMap<>();
```

Comment:
This can be inlined in the one test that uses it

Reply:
Not really cause the dofn would loose the reference and therefore the test would fail, is there a way with the direct runner to keep references accross the pipeline execution?

Reply:
right, closure stuff and serde cycles are a pain, you're right. However, you can write this like is done in `reusePipelineSucceeds` - declare the field you're using right next to the single test that uses it, only interact with it in that test. This makes it much easier to read the test. Additionally, it means that you can do a simple initialization within the test instead of having to introduce `@Before` and `@After` methods and new test rules.